recipes/diligent-core/all/conanfile.py

```diff
@@ -0,0 +1,169 @@
+    license = ("Apache 2.0")
+    topics = ("graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "spirv_cross_namespace": "ANY",
+    }
+    default_options = {
+        "fPIC": True,
+        "spirv_cross_namespace": "spirv_cross",
+    }
+    generators = "cmake_find_package", "cmake"
+    _cmake = None
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    short_paths = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "6",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("Visual Studio build with MT runtime is not supported")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package_id(self):
+        if self.settings.compiler == "Visual Studio":
+            if "MD" in self.settings.compiler.runtime:
+                self.info.settings.compiler.runtime = "MD/MDd"
+            else:
+                self.info.settings.compiler.runtime = "MT/MTd"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def requirements(self):
+        self.requires("opengl/system")
+
+        self.requires("libjpeg/9d")
+        self.requires("libtiff/4.3.0")
+        self.requires("zlib/1.2.11")
+        self.requires("libpng/1.6.37")
+
+        self.requires("spirv-cross/cci.20210930")
+        self.options["spirv-cross"].namespace = self.options.spirv_cross_namespace
```

Comment:
I'd of expected this in validate?

Reply:
do not override dependencies options in requirements

Reply:
How can I get an option from a dependency?

Reply:
Best answer I know of https://github.com/conan-io/conan-center-index/pull/7407#discussion_r723086207